Dashboard/app/js/lib/components/devices/AssignmentsEditView/index.jsx

```diff
@@ -38,115 +32,61 @@ export default class AssignmentsEdit extends React.Component {
+              id="assignmentName"
+              placeholder={strings.assignmentNamePlaceholder}
+              value={this.state.data.assignmentName}
+              onChange={this.onChangeState}
+            />
+            {/* <span className="infoText">0 datapoints / 20k assigned</span> */}
+          </h3>
+        </div>
 
-    return selectedDevices;
-  };
+        <button onClick={this.onSubmit} type="button" className="standardBtn">
+          {strings.saveAssignment}
+        </button>
+      </div>
+    );
+  }
 
   render() {
-    const { strings, actions, data } = this.props;
-    // const numberOfDevices = this.getNumberOfSelectedDevices();
+    const contextData = {
+      strings: this.props.strings,
+      actions: {
+        ...this.props.actions,
+        onInputChange: this.onChangeState,
+      },
+      data: this.props.data,
+      inputValues: this.state.data,
+    };
 
     return (
       <div className="assignments-edit">
         {/* topbar */}
-        <div className="assignment-topbar">
-          <div className="assignment-name">
-            <button
-              type="button"
-              className="go-back"
-              onClick={actions.cancelEditSurveyAssignment}
-            >
-              <i className="fa fa-arrow-left" />
-            </button>
-
-            <h3>
-              <input
-                type="text"
-                id="assignmentName"
-                placeholder={strings.assignmentNamePlaceholder}
-                value={this.state.data.assignmentName}
-                onChange={this.onChangeState}
-              />
-              {/* <span className="infoText">0 datapoints / 20k assigned</span> */}
-            </h3>
-          </div>
-
-          <button onClick={this.onSubmit} type="button" className="standardBtn">
-            {strings.saveAssignment}
-          </button>
-        </div>
+        {this.renderTopBar()}
 
         <div className="assignment-body">
-          <div className="assignment-sidebar">
-            <ul>
-              <li className={this.state.currentTab === 'FORMS' ? 'active' : ''}>
-                <button type="button" onClick={() => this.onChangeTab('FORMS')}>
-                  Forms
-                </button>
-              </li>
-
-              <li
-                className={this.state.currentTab === 'DEVICES' ? 'active' : ''}
-              >
-                <button
-                  type="button"
-                  onClick={() => this.onChangeTab('DEVICES')}
-                >
-                  Devices
-                </button>
-              </li>
-            </ul>
-          </div>
-
-          <div className="assignment-main">
-            <FormSection
-              actions={actions}
-              strings={strings}
-              data={data}
-              inputValues={this.state.data}
-              onInputChange={this.onChangeState}
-            />
-          </div>
+          <AssignmentContext.Provider value={contextData}>
```

Comment:
I would have though that the context here has two parts.. the SideBar and the MainSection

Reply:
The data is not for the sidebar. It's for the whole component. So it can be accessed from any component since they're lots of deeply nested components.